### PR TITLE
refactor(settings): change HathorManager to use injected settings

### PR DESCRIPTION
**Pull request by glevco**
### Motivation

Continuing with the settings-related refactors, this PR updates `HathorManager` so it starts using injected settings, instead of global. This refactor is necessary for the Feature Activation mandatory signaling implementation.

### Acceptance Criteria

- Add `settings` attribute to `HathorManager`, removing the module level settings
- Move `DEFAULT_CAPABILITIES` from module level to `HathorManager.get_default_capabilities()`
- Update `Builder` and `CliBuilder` accordingly

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: hathor/manager.py [hunk 1]
```diff
@@ -130,6 +123,7 @@ def __init__(self,
                 'Either enable it, or use the reset-event-queue CLI command to remove all event-related data'
             )
 
+        self.settings = settings
```
Comment (jansegre) [inline hathor/manager.py:126]: Other classes use `self._settings` should we do this here too?

File: hathor/manager.py [hunk 2]
```diff
@@ -233,6 +227,13 @@ def __init__(self,
         self.lc_check_sync_state.clock = self.reactor
         self.lc_check_sync_state_interval = self.CHECK_SYNC_STATE_INTERVAL
 
+    def get_default_capabilities(self) -> list[str]:
```
Comment (msbrogli) [inline hathor/manager.py:230]: Missing docstring.

**Review by jansegre — APPROVED** [review]

**Review by msbrogli — DISMISSED** [review]

File: hathor/manager.py (hunk 2 above)
Comment (glevco) [reply hathor/manager.py:230]: Done in https://github.com/HathorNetwork/hathor-core/pull/787/commits/914dd775149b55db26cf5e9d0e8e29d3ef118681

File: hathor/manager.py (hunk 1 above)
Comment (glevco) [reply hathor/manager.py]: Makes sense, done in https://github.com/HathorNetwork/hathor-core/pull/787/commits/748afab81c05dbd564a00c5274bacaec0526cc5e

File: hathor/manager.py (hunk 2 above)
Comment (msbrogli) [reply hathor/manager.py:230]: Thanks!

**Review by msbrogli — APPROVED** [review]

**Pull request merged**